xbmc/Application.cpp

```diff
@@ -3173,7 +3173,8 @@ void CApplication::OnPlayBackStarted(const CFileItem &file)
   CLog::LogF(LOGDEBUG,"CApplication::OnPlayBackStarted");
 
   // check if VideoPlayer should set file item stream details from its current streams
-  if (file.GetProperty("get_stream_details_from_player").asBoolean())
+  if (file.GetProperty("get_stream_details_from_player").asBoolean() || file.IsDiscImage() ||
+      file.IsDVDFile() || file.IsBDFile() || file.IsBluray() || file.IsDVD() || file.IsOnDVD())
```

Comment:
Perhaps add: "|| !file.HasVideoInfoTag() || !file.GetVideoInfoTag()->HasStreamDetails() ?

Reply:
OMG. Looks horrible.

Reply:
Well, perhaps the complete if statement can simply be replaced with "if (!file.HasVideoInfoTag() || !file.GetVideoInfoTag()->HasStreamDetails()) ?